src/context/history-context/history-context.js

```diff
@@ -0,0 +1,126 @@
+import { useContext, useState, createContext } from "react";
+import { useEffect } from "react";
+import axios from "axios";
+import { useToast } from "../toast-context/toast-context";
+
+const HistoryContext = createContext();
+
+const useHistory = () => useContext(HistoryContext);
+
+function HistoryProvider({ children }) {
+  const encodedToken = localStorage.getItem("smashTubeToken");
+  const [history, setHistory] = useState([]);
+  const { dispatch } = useToast();
+
+  useEffect(() => {
+    (async function () {
+      if (encodedToken) {
+        try {
+          const historyResponse = await axios.get("/api/user/history", {
+            headers: {
+              authorization: encodedToken,
+            },
+          });
+
+          if (historyResponse.status === 200) {
```

Comment:
api success status can be between 200 and 299 make condition >200 and less than 299

Reply:
will look into this :)